File: 3sem/4lab/Banks/Exceptions/InvalidBankOperationException.cs
namespace Banks.Exceptions;

public class InvalidBankOperationException : BanksDomainException
{
    private InvalidBankOperationException(string? message)
        : base(message) { }

    public static InvalidBankOperationException OnAddExistingClient()
        => new InvalidBankOperationException($"Bank already has this client");

    public static InvalidBankOperationException OnAddAccountToUnregisteredClient()
        => new InvalidBankOperationException($"Unable to create bank account to unregistered account");
}

File: 4sem/1lab/src/main/java/BanksRemake/Domain/Exceptions/InvalidBankOperationException.java
package banksRemake.domain.exceptions;

public class InvalidBankOperationException extends BanksRemakeException {
    private InvalidBankOperationException(String message) {
        super(message);
    }

    public static InvalidBankOperationException onAddExistingClient() {
        return new InvalidBankOperationException("Bank already has this client");
    }

    public static InvalidBankOperationException onAddAccountToUnregisteredClient() {
        return new InvalidBankOperationException("Unable to create bank account to unregistered account");
    }
}
